### PR TITLE
Collection of small fixes and improvements that do not merit an own PR each

### PR DESCRIPTION
**Pull request by hulpke**
Partial backport of my work branch to master

* Performance improvement to automorphism group/group isomorphism *
- Clean away the messy EFactors function in `DataAboutSimpleGroup`
- Store Solvability of extensions, if known
- Have RankAction test transitivity (was requested)
* FIX: Improvements in how MTC deletes superfluous generators -- avoid error message in ghomomorphisms for FpGroups. *
* Performance improvement in `IntermediateSubgroups` if the smaller subgroup has many orbits *

(The ones with * merit mentioning in the release notes)

### CODE REVIEW

**Comment by wilfwilson** [comment]
Thanks @hulpke. I don’t have time to review at the moment, but I’ll note that GitHub renders `-` and `*` as the same bullet point icon, so it’s not obvious which ones you mean for release notes.

**Comment by hulpke** [comment]
Thank you @wilfwilson (and I was so proud of using two different bullets... :-( )

File: lib/ghomfp.gi [hunk 1]
```diff
@@ -871,7 +871,10 @@ local aug,w,p,pres,f,fam,G,trace;
 
     pres:=NEWTC_PresentationMTC(aug,0,nam);
   fi;
-  Assert(0,Length(GeneratorsOfPresentation(pres))=Length(gens));
+  # check that we have the exat generators as we want and no rearrangement
```
Comment (wilfwilson) [inline lib/ghomfp.gi:874]:
```suggestion
  # check that we have the exact generators as we want and no rearrangement
```

**Review by wilfwilson — APPROVED** [review]
Looks sensible to me.

**Pull request merged**